### PR TITLE
Add: audit event to GET /roles/:account/:kind/*identifier?memberships

### PR DESCRIPTION
**Pull request by yoavgeva**
### Desired Outcome

This PR adds an audit log message for list members using the API endpoint `GET /roles/{account}/{kind}/{identifier}?memberships`

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- Example audit message

**Audit Example:**
```
<86>1 2022-12-29T21:56:07.426Z - conjur 74681 memberships [auth@43868 user="cucumber:user:admin"][subject@43868 role="cucumber:user:alice"][client@43868 ip="127.0.0.1"][action@43868 result="success" operation="list"] cucumber:user:admin successfully listed memberships with parameters: {:account=>"cucumber", :kind=>"user"}
```

### Connected Issue/Story

Resolves [ONYX-26177](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-26177)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: app/controllers/roles_controller.rb [hunk 1]
```diff
@@ -54,8 +54,14 @@ def all_memberships
   # +params[:filter]+ and +params[:count]+ are handled as for +#all_memberships+
   #
   def direct_memberships
-    memberships = filtered_roles(role.direct_memberships_dataset(filter_params), membership_filter)
-    render_dataset(memberships)
+    filter = membership_filter
```
Comment (aloncarmel111) [inline app/controllers/roles_controller.rb:57]: not need to create new variable for this

**Review by jvanderhoof — APPROVED** [review]
Looks good.  Thanks for getting this functional gap filled!

File: app/controllers/roles_controller.rb [hunk 2]
```diff
@@ -225,4 +230,38 @@ def count_only?
   def count_payload(dataset)
     { count: dataset.count }
   end
+
+  def audit_memberships_success(filter)
+    additional_params = %i[account count search kind filter]
+    options = params.permit(*additional_params)
+                    .slice(*additional_params).to_h.symbolize_keys
```
Comment (jvanderhoof) [inline app/controllers/roles_controller.rb:237]: We should also strip the extra `slice` here:

```suggestion
                    .slice(*additional_params).to_h.symbolize_keys
```
```suggestion
                    .to_h.symbolize_keys
```

File: app/controllers/roles_controller.rb [hunk 3]
```diff
@@ -225,4 +230,38 @@ def count_only?
...
+        subject: options,
+        success: true
+      )
+    )
+  end
+
+  def audit_memberships_failure(filter, err)
+    additional_params = %i[account count search kind filter]
+    options = params.permit(*additional_params)
+                    .slice(*additional_params).to_h.symbolize_keys
```
Comment (jvanderhoof) [inline app/controllers/roles_controller.rb:253]: Same here:

```suggestion
                    .to_h.symbolize_keys
```

Comment (yoavgeva) [reply app/controllers/roles_controller.rb]: @jvanderhoof Thanks on the comment.

**Review by jvanderhoof — APPROVED** [review]

**Review by jvanderhoof — APPROVED** [review]

**Pull request merged**